client/src/review/review.jsx

```diff
@@ -3,41 +3,84 @@ import ReviewList from './reviewList.jsx';
 import RatingBreakdown from './ratingBreakdown.jsx';
 import reviewData from '../../../reviewSampleData.json';
 import ProductBreakdown from './productBreakdown.jsx';
+import AddReviewPopup from './addReviewPopup.jsx';
 import $ from 'jquery';
-import { Link } from 'react-router-dom';
 class Review extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { data: reviewData };
-    this.getReviewsData();
+    this.state = {
+      data: reviewData,
+      sort: 'relevance',
+      filters: [],
+      popup: false,
+    };
+    this.getReviewsData(props.id);
+    this.addStarFilter = this.addStarFilter.bind(this);
+    this.removeStarFilter = this.removeStarFilter.bind(this);
+    this.updateSortType = this.updateSortType.bind(this);
+    this.togglePopup = this.togglePopup.bind(this);
+
+  }
+
+  addStarFilter(e) {
+    let filters = this.state.filters;
+    let newFilter = e.target.text.split(' ')[0];
+    if (!filters.includes(newFilter)) {
+      filters.push(newFilter);
+      this.setState({ filters });
+      console.log(this.state.filters);
+    }
+  }
+
+  removeStarFilter(e) {
+    let filters = this.state.filters;
+    let tbdFilter = e.target.name.split(' ')[0];
```

Comment:
Does this variable name need to be updated? or is it still tbd

Reply:
oh, it means to-be-deleted, this is related to a function of  adding and removing filters for reviews.

> Does this variable name need to be updated? or is it still tbd